### PR TITLE
fix: divide auth logic, fix init time

### PR DESCRIPTION
**Pull request by subbotnik16**

### CODE REVIEW

File: src/navigation/hooks/useMagicAuth.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+import {magicLink} from '@services/magicLink';
```
Comment (pisarevkir) [inline src/navigation/hooks/useMagicAuth.ts:3]: The general idea is to keep all the business logic in sagas, `hook` folder is for general non-business stuff (UI animations, listeners and so on). Moreover we already have 'Auth' module, lets move that logic there?

We can add a common event, like "APP_LOADED" that would be dispatched on app startup and we can trigger all the work related to the app start on that event. So APP_LOADED -> module Auth starts saga that checks if user is authorized -> saves data / result in the store -> the UI layer only uses selectors to grab the data from the store (either user logged or not, user data and so on)

Comment (victor-oancea-leftclick-io) [reply src/navigation/hooks/useMagicAuth.ts:3]: +1

**Review by victor-oancea-leftclick-io — DISMISSED** [review]

File: src/screens/Home/index.tsx [hunk 2]
```diff
@@ -10,8 +11,9 @@ import {useDispatch, useSelector} from 'react-redux';
 export const Home = () => {
   const {email} = useSelector((state: RootState) => state.auth.userData);
   const dispatch = useDispatch();
-  const logOutPress = () => {
+  const logOutPress = async () => {
     dispatch(AuthActions.SIGN_OUT.START.create());
+    await magicLink.logoutUser();
```
Comment (pisarevkir) [inline src/screens/Home/index.tsx:16]: logout logic should be also in sagas

File: src/store/modules/AppCommon/reducer/index.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+function reducer(state = INITIAL_STATE, action: Actions): State {
+  return produce(state, draft => {
+    switch (action.type) {
+      case AppCommonActions.APP_LOADED.STATE.type:
+        draft.isAppLoaded = true;
+        break;
+      case AuthActions.SIGN_OUT.SUCCESS.type: {
+        return {
+          ...INITIAL_STATE,
```
Comment (pisarevkir) [inline src/store/modules/AppCommon/reducer/index.ts:30]: no need to reset isAppLoaded - it should be fired only once, when the app is actually loaded

File: src/store/modules/Auth/selectors/index.ts [hunk 4]
```diff
@@ -19,3 +19,13 @@ export const isSignUpCompletedSelector = createSelector(
     );
   },
 );
+
+export const userDataSelector = createSelector(
+  authRootSelector,
```
Comment (pisarevkir) [inline src/store/modules/Auth/selectors/index.ts:24]: `authRootSelector` and usage of `reselect` doesn't make sense here, it doesn't give any performance boost - 

```
export const getInitSelector = (state: RootState) => state.auth.userData;
```

File: src/services/magicLink.ts [hunk 5]
```diff
@@ -11,21 +11,12 @@ export const magic = new Magic(ENV.MAGIC_LINK_KEY, {
 class MagicLink {
```
Comment (pisarevkir) [inline src/services/magicLink.ts:11]: If we keep the business logic in sagas, we don't need this kind of Services - UI should dispatch `loginUser` and `socialLogin` and so on and `magic.*` calls should happen in sagas

File: src/store/modules/Auth/reducer/index.ts [hunk 6]
```diff
@@ -45,18 +46,14 @@ const INITIAL_STATE: State = {
   initialization: true,
```
Comment (pisarevkir) [inline src/store/modules/Auth/reducer/index.ts:46]: may be name it something like `isInitialized` or `isUserInitialized`? That would be the same pattern as we apply to other flags? I was first confused when i saw `const initialization = useSelector(getInitSelector);`

BTW - no need to add `get` to selector names - they are always `get`, so `getInitSelector` -> `userInitializedSelector`

File: src/store/modules/AppCommon/selectors/index.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+import {RootState} from '@store/rootReducer';
+import {createSelector} from 'reselect';
+
+const appCommonRootSelector = (state: RootState) => state.appCommon;
```
Comment (pisarevkir) [inline src/store/modules/AppCommon/selectors/index.ts:6]: same as for other selectors - no need to use `reselect` in this case, 
```
export const isAppLoadedSelector = (state: RootState) => state.appCommon.isAppLoaded;
```

File: src/store/modules/AppCommon/reducer/index.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+const persistConfig = {
+  key: 'appCommon',
+  storage: AsyncStorage,
+  timeout: 120000,
+  blacklist: ['isAppLoaded'],
+};
+
+export const appCommonReducer = persistReducer(persistConfig, reducer);
```
Comment (pisarevkir) [inline src/store/modules/AppCommon/reducer/index.ts:44]: no need to persist reducer if we don't keep any data

**Review by pisarevkir — APPROVED** [review]

**Review by victor-oancea-leftclick-io — APPROVED** [review]

**Pull request merged**